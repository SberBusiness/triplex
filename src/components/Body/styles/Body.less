@import 'src/styles/style';

.body {
    position: relative;
    padding: @ufs-theme-page-padding-y @ufs-theme-page-padding-x;
    display: flex;

    .bodyInner {
        position: relative;
        flex-grow: 1;
        width: 100%;
    }

    &.withWhiteBackground {
        background: var(--triplex-Body-Background_White);
    }

    &.withGrayBackground {
        background: var(--triplex-Body-Background_Gray);
    }

    &.withDarkGrayBackground {
        background: var(--triplex-Body-Background_Darkgray);
    }
}

@media (max-width: @screen-sm-max) {
    .body {
        padding: @page-padding-sm;
        overflow-x: auto;
    }
}
