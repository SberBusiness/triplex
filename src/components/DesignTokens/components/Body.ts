import {TDesignTokenValue, TDesignTokenValues} from '@sberbusiness/triplex/components/DesignTokens/types/DesignTokenTypes';

// Название токенов компонента Body.
export const designTokensComponentsBodyKeys = ['Background_White', 'Background_Gray', 'Background_Darkgray'] as const;
// Тип, содержащий названия токенов компонента Body.
export type TDesignTokensComponentsBodyKeys = (typeof designTokensComponentsBodyKeys)[number];
// Тип, содержащий названия токенов компонента Body и их значения.
export type TDesignTokensComponentsBodyValue = Record<TDesignTokensComponentsBodyKeys, TDesignTokenValue>;
// Тип, содержащий названия токенов компонента Body и их значения в светлой и темной теме.
export type TDesignTokensComponentsBodyValues = Record<TDesignTokensComponentsBodyKeys, TDesignTokenValues>;
// Тип локальных токенов компонента Body.
export type TDesignTokensComponentsBody = {Body: TDesignTokensComponentsBodyValue};

// Токены компонента Body в светлой и темной темах.
export const Body_Tokens: TDesignTokensComponentsBodyValues = {
    Background_Darkgray: [{ref: 'Neutral.900'}, {ref: 'Basic.700'}], // var(--triplex-Body-Background_Darkgray)
    Background_Gray: [{ref: 'Neutral.100'}, {ref: 'Basic.700'}], // var(--triplex-Body-Background_Gray)
    Background_White: [{ref: 'Basic.100'}, {ref: 'Basic.900'}], // var(--triplex-Body-Background_White)
};
