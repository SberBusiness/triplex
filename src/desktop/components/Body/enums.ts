export enum EBodyBackgroundColor {
    WHITE = 'white',
    GRAY = 'gray',
    DARKGRAY = 'darkgray',
}
