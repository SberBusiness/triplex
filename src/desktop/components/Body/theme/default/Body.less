@import '../../../../common/theme/default';

.body {
    position: relative;
    padding: @ufs-theme-page-padding-y @ufs-theme-page-padding-x;
    display: flex;

    .bodyInner {
        position: relative;
        flex-grow: 1;
        width: 100%;
    }

    &.withWhiteBackground {
        background: @color-snowy;
    }

    &.withGrayBackground {
        background: @color-asphalt-05;
    }

    &.withDarkGrayBackground {
        background: @color-asphalt-01;
    }
}

@media (max-width: @screen-sm-max) {
    .body {
        padding: @page-padding-sm;
        overflow-x: auto;
    }
}
