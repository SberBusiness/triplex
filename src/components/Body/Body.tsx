import React from 'react';
import {classnames} from '@sberbusiness/triplex/utils/classnames/classnames';
import {EBodyBackgroundColor} from '@sberbusiness/triplex/components/Body/enums';

/** Свойства компонента Body. */
export interface IBodyProps extends React.HTMLAttributes<HTMLDivElement> {
    /** Цвет фона. */
    backgroundColor?: EBodyBackgroundColor;
}

/**
 * Компонент Body, используется как один из детей(Header, Body и Footer) компонента Page.
 * Используется как контейнер для контента страницы. Имеет дефолтные отступы со всех сторон.
 */
export const Body: React.FC<IBodyProps> = ({children, className, backgroundColor = EBodyBackgroundColor.WHITE, ...htmlDivAttributes}) => (
    <div
        className={classnames('cssClass[body]', className, {
            'cssClass[withDarkGrayBackground]': backgroundColor === EBodyBackgroundColor.DARKGRAY,
            'cssClass[withGrayBackground]': backgroundColor === EBodyBackgroundColor.GRAY,
            'cssClass[withWhiteBackground]': backgroundColor === EBodyBackgroundColor.WHITE,
        })}
        {...htmlDivAttributes}
        data-tx={process.env.npm_package_version}
    >
        <div className="cssClass[bodyInner]">{children}</div>
    </div>
);

Body.displayName = 'Body';
