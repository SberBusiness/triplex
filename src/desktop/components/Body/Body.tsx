import {classnames} from '@sberbusiness/triplex/common/utils/classnames/classnames';
import {EBodyBackgroundColor} from '@sberbusiness/triplex/desktop/components/Body/enums';
import * as React from 'react';

/**
 * Свойства компонента.
 *
 * @prop {boolean} [grayBackground] Добавляет серый фон для body.
 */
export interface IBodyProps extends React.HTMLAttributes<HTMLDivElement> {
    backgroundColor?: EBodyBackgroundColor;
}

/**
 * Компонент Body, используется как один из детей(Header, Body и Footer) компонента Page.
 * Используется как контейнер для контента страницы. Имеет дефолтные отступы со всех сторон.
 */
export const Body: React.FC<IBodyProps> = ({children, className, backgroundColor = EBodyBackgroundColor.WHITE, ...htmlDivAttributes}) => (
    <div
        className={classnames('cssClass[body]', className, {
            'cssClass[withGrayBackground]': backgroundColor === EBodyBackgroundColor.GRAY,
            'cssClass[withDarkGrayBackground]': backgroundColor === EBodyBackgroundColor.DARKGRAY,
            'cssClass[withWhiteBackground]': backgroundColor === EBodyBackgroundColor.WHITE,
        })}
        {...htmlDivAttributes}
    >
        <div className="cssClass[bodyInner]">{children}</div>
    </div>
);

Body.displayName = 'Body';
