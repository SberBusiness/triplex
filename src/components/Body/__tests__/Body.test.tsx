import React from 'react';
import renderer from 'react-test-renderer';
import {Body} from '../Body';

describe('Body', () => {
    beforeEach(() => {
        allure.feature('Body');
    });

    it('renders correctly', () => {
        const tree = renderer.create(<Body>Text</Body>).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
