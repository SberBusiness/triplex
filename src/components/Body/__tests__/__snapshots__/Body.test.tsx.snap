// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Body renders correctly 1`] = `
<div
  className="cssClass[body] cssClass[withWhiteBackground]"
  data-tx="10.0.0"
>
  <div
    className="cssClass[bodyInner]"
  >
    Text
  </div>
</div>
`;
